infer/src/IR/Procdesc.re

```diff
@@ -519,3 +519,35 @@ let node_set_succs_exn pdesc (node: Node.t) succs exn =>
     set_succs_exn_base node' [exit_node] exn
   | _ => set_succs_exn_base node succs exn
   };
+
+let get_loop_heads pdesc => {
```

Comment:
(1) Yes.
(2) Yes if `Node.get_succs` returns edges of `break` / `continue` / `return`.

The classical algorithm for computing loop heads in compilers is not good for setting widening points because it does not catch all cycles. For example,
```c
if(cond){
a:
  goto b;
}else{
b:
  goto a;
}
```
there is a cycle between `a` and `b`, so one of them should be set as a widening point. However, they cannot be loop heads by the classical algorithm because `a` and `b` do not dominate each other.

While the classical algorithm finds loop heads by back edges to dominators, our algorithm finds all nodes that have back edges when drawing depth-first spanning tree.

(3) You are right. We added `Node.get_exn` to find exceptional control-flow.